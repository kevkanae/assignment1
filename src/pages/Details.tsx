import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  Typography,
} from "@mui/material";
import { useQuery } from "react-query";
import { useParams } from "react-router-dom";
import CountryCard from "../components/CountryCard";
import { ICountryDetails } from "../interfaces/CountryDetails";
import { getCountryDetailsAPI } from "../services/GetCountryDetails";
import { flexCol } from "../styles/flex";

const Details = () => {
  let params = useParams();
  const { data, isFetched, isError, error } = useQuery<
    ICountryDetails[],
    Error
  >(["product", params.countryName], () =>
    getCountryDetailsAPI(params.countryName as string)
  );

  if (isError) {
    return (
      <>
        <Box>There was an Error: {error}</Box>
      </>
    );
  } else if (isFetched && !isError && data) {
    return (
      <Box
        sx={{
          ...flexCol,
          justifyContent: "center",
          height: "100vh",
          width: "100vw",
        }}
      >
        <CountryCard data={data[0] as ICountryDetails} />
      </Box>
    );
  } else {
    return <Box>Loading...</Box>;
  }
};

export default Details;
