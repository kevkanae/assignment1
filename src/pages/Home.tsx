import { Box, TextField, Button } from "@mui/material";
import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { buttonStyles } from "../styles/ButtonStyles";
import { countryNameStyles } from "../styles/InputStyles";

const Home = () => {
  const navigate = useNavigate();
  const [country, setCountry] = useState<string>("");

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) =>
    setCountry(e.currentTarget.value);

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
        height: "100vh",
        width: "100vw",
        backgroundColor: "#0f1a2b",
      }}
    >
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          alignItems: "center",
          justifyContent: "space-evenly",
          boxShadow: 2,
          height: "30vh",
          width: "35vw",
          borderRadius: 2,
          p: "1.4rem",
          color: "#f2ffff",
          backgroundColor: "#14253d",
        }}
      >
        <TextField
          variant="outlined"
          label="Country Name"
          onChange={handleChange}
          value={country}
          sx={{
            ...countryNameStyles,
          }}
        />
        <Button
          variant="outlined"
          sx={{
            ...buttonStyles,
          }}
          disabled={country === ""}
          onClick={() => navigate(`/${country}`)}
        >
          Submit
        </Button>
      </Box>
    </Box>
  );
};

export default Home;
