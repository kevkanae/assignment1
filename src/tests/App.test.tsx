import { render } from "@testing-library/react";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import Details from "../pages/Details";
import Home from "../pages/Home";

describe("do my routers", () => {
  test("work correctly", () => {
    let view = render(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/:countryName" element={<Details />} />
        </Routes>
      </MemoryRouter>
    );

    expect(view).toMatchSnapshot();
  });
});
