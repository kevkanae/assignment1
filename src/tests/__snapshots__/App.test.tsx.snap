// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`do my routers work correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1do5ga1"
      >
        <div
          class="MuiBox-root css-yrjpw7"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-ki6ctg-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r0:"
              id=":r0:-label"
            >
              Country Name
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Country Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-18g54ta-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1do5ga1"
    >
      <div
        class="MuiBox-root css-yrjpw7"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-ki6ctg-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Country Name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Country Name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled  css-18g54ta-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
